### PR TITLE
Add permission filter for list collaborators

### PR DESCRIPTION
**Pull request by Sairaviteja27**
Fixes https://github.com/google/go-github/issues/2651

### CODE REVIEW

**Review by MytsYurii — APPROVED** [review]
looks good

File: github/repos_collaborators.go [hunk 1]
```diff
@@ -23,6 +23,13 @@ type ListCollaboratorsOptions struct {
 	// Default value is "all".
 	Affiliation string `url:"affiliation,omitempty"`
 
+	// Permission specifies how collaborators should be filtered by the permissions they have on the repository.
+	// Possible values are:
+	// pull, triage, push, maintain, admin
```
Comment (gmlewis) [inline github/repos_collaborators.go:28]: For consistency, let's please put these in double quotes.

```suggestion
	// "pull", "triage", "push", "maintain", "admin"
```

File: github/repos_collaborators.go [hunk 2]
```diff
@@ -23,6 +23,13 @@ type ListCollaboratorsOptions struct {
...
+	//
+	// If not specified, all collaborators will be returned
```
Comment (gmlewis) [inline github/repos_collaborators.go:30]: nit:

```suggestion
	// If not specified, all collaborators will be returned.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @Sairaviteja27 !
Just a couple tiny tweaks, please.

**Pull request merged**